test/e2e/usersignup_test.go

```diff
@@ -284,6 +284,39 @@ func (s *userSignupIntegrationTest) TestUserResourcesCreatedWhenOriginalSubIsSet
 	VerifyResourcesProvisionedForSignup(s.T(), s.Awaitilities, userSignup, "deactivate30", "base")
 }
 
+func (s *userSignupIntegrationTest) TestUserResourcesUpdatedWhenPropagatedClaimsModified() {
+	hostAwait := s.Host()
+
+	// given
+	hostAwait.UpdateToolchainConfig(s.T(), testconfig.AutomaticApproval().Enabled(true))
+
+	// when
+	userSignup, _ := NewSignupRequest(s.Awaitilities).
+		Username("test-user-resources-updated").
+		Email("test-user-resources-updated@redhat.com").
+		UserID("43215432").
+		AccountID("ppqnnn00099").
+		EnsureMUR().
+		RequireConditions(wait.ConditionSet(wait.Default(), wait.ApprovedAutomatically())...).
+		Execute(s.T()).Resources()
+
+	// then
+	VerifyResourcesProvisionedForSignup(s.T(), s.Awaitilities, userSignup, "deactivate30", "base")
+
+	// Update the UserSignup
+	userSignup, err := hostAwait.UpdateUserSignup(s.T(), userSignup.Name, func(us *toolchainv1alpha1.UserSignup) {
+		// Modify the user's AccountID
+		us.Spec.IdentityClaims.AccountID = "nnnbbb111234"
+	})
+	require.NoError(s.T(), err)
+
+	// Confirm the AccountID is updated
+	require.Equal(s.T(), "nnnbbb111234", userSignup.Spec.IdentityClaims.AccountID)
```

Comment:
Hm... Doesn't it just test the `hostAwait.UpdateUserSignup()` function? :)
Shouldn't you check the `UserAccount.Spec.PropogatedClaims` have been updated instead (not the UserSignup!)? And maybe the `Mur.Spec.PropogatedClaims` too if we do want to check all intermediate resources ;)

Reply:
The `UserAccount.Spec.PropagatedClaims` is checked in `VerifyResourcesProvisionedForSignup` (technically in `VerifyUserRelatedResources` in `user_assertions.go`) on line 317, just below this code.  The MUR also undergoes some verification, but not its PropagatedClaims - however since the `UserAccount` is verified in this very same function, then it's impossible for MUR to have different values as it is the source of the property values propagated to `UserAccount` in the first place.

Reply:
Ok. Sounds good then.

Reply:
for better readability, it would be nice to add a specific wait (before `VerifyResourcesProvisionedForSignup`) that would verify that the value is propagated down to the UserAccount.  I know that `VerifyResourcesProvisionedForSignup` checks that the UserAccount has the same values of the propagatedClaims as is in UserSignup CR, but having it here specified explicitly as well would be better for reading the code in the future